src/fs_event_wrap.cc

```diff
@@ -156,7 +160,36 @@ void FSEventWrap::OnEvent(uv_fs_event_t* handle, const char* filename,
   };
 
   if (filename != nullptr) {
-    argv[2] = OneByteString(env->isolate(), filename);
+    size_t len = strlen(filename);
+    // If encoding is BUFFER, return a Buffer.
+    // If encoding is UCS2, attempt to decode as UCS2 or report an error.
+    // Otherwise, Encode as the requested type, with UTF8 as the fallback.
+    if (wrap->encoding_ == BUFFER) {
+      argv[2] = Buffer::Copy(env, filename, len).ToLocalChecked();
+    } else if (wrap->encoding_ == UCS2) {
+      // Note: this is lossy. If the filename is not aligned at 2 bytes,
+      // the last byte in the filename will be silently dropped in the 
+      // copy.
+      const uint16_t* buf;
+      bool release = false;
+      len /= 2;
+      if ((len % sizeof(*buf) != 0)) {
+        uint16_t* copy = new uint16_t[len];
+        memcpy(&buf, &filename, len);
+        buf = copy;
+        release = true;
+      } else {
+        buf = reinterpret_cast<const uint16_t*>(filename);
```

Comment:
This is unsound; `filename` doesn't have to be aligned on a two-byte boundary.  I'd write it like this:
```c++
const size_t size = strlen(filename);
if (size % 2 != 0) /* Error */;
std::vector<uint16_t> vec(size / 2);
for (size_t i = 0, k = 0; i < size; i += 2, k += 1) {
  const uint8_t hi = static_cast<uint8_t>(filename[i + 0]);
  const uint8_t lo = static_cast<uint8_t>(filename[i + 1]);
  vec[k] = static_cast<uint16_t>(hi) * 256 + lo;
}
argv[2] =
    vec.empty() ? String::Empty(env->isolate())
    : StringBytes::Encode(env->isolate(), &vec[0], vec.size());
```
Swap `hi` and `lo` for LE encoding.  There are probably enough places where we do similar fix-ups that it pays to turn it into a helper function.